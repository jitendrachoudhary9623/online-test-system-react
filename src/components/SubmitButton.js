import React,{Component} from "react";

export default class SubmitButton extends Component{

    render(){
       return (
        <div className="row">
        <button className="col s12 m-2 center-align btn waves-effect waves-light blue darken-1" type="submit" name="action">Submit
            <i className="material-icons right"></i>
        </button>
    </div>
       ) 
    }
}