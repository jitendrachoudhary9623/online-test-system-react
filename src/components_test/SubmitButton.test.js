import React, { Component } from 'react';
import SubmitButton from "../components/client_components/SubmitButton"
import { shallow } from 'enzyme'


describe('UI Test for Submit Button', () => {

    it('Submit Button to be defined', () => {
        expect(SubmitButton).toBeDefined();
    });

    it('Submit Button  should render correctly', () => {
        const snapshot = shallow(
          <SubmitButton/>
        );
        expect(snapshot).toMatchSnapshot();
      });
})