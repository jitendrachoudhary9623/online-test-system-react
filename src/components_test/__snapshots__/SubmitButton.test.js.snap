exports[`UI Test for Submit Button Submit Button  should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubmitButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="col s12 m-2 center-align btn waves-effect waves-light blue darken-1"
        name="action"
        type="submit">
        Submit
        <i
                className="material-icons right" />
</button>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Submit",
          <i
            className="material-icons right" />,
        ],
        "className": "col s12 m-2 center-align btn waves-effect waves-light blue darken-1",
        "name": "action",
        "type": "submit",
      },
      "ref": null,
      "rendered": Array [
        "Submit",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "material-icons right",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="col s12 m-2 center-align btn waves-effect waves-light blue darken-1"
          name="action"
          type="submit">
          Submit
          <i
                    className="material-icons right" />
</button>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Submit",
            <i
              className="material-icons right" />,
          ],
          "className": "col s12 m-2 center-align btn waves-effect waves-light blue darken-1",
          "name": "action",
          "type": "submit",
        },
        "ref": null,
        "rendered": Array [
          "Submit",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "material-icons right",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
